borsh/src/ser/mod.rs

```diff
@@ -201,6 +201,41 @@ impl BorshSerialize for String {
     }
 }
 
+#[cfg(feature = "bigdecimal")]
+impl BorshSerialize for bigdecimal::BigDecimal {
+    #[inline]
+    fn serialize<W: Write>(&self, writer: &mut W) -> Result<()> {
+        let (bigint, exponent) = self.as_bigint_and_exponent();
+        bigint.serialize(writer)?;
+        exponent.serialize(writer)
+    }
+}
+
+#[cfg(feature = "bigdecimal")]
+impl BorshSerialize for bigdecimal::num_bigint::BigInt {
+    #[inline]
+    fn serialize<W: Write>(&self, writer: &mut W) -> Result<()> {
+        let (sign, mut data) = self.to_bytes_le();
+        while data.last() == Some(&0) {
+            data.pop();
+        }
```

Comment:
I don't believe this is necessary based on the implementation, seems we can just write a test for this (padded BigInt and ensuring serialized value doesn't have trailing 0s) and not have this logic. The only check we need is to validate on deserialization it only allows one canonical format and doesn't have padded 0 bytes

Reply:
It actually has 0, at least for 0 itself, the array is not empty.

Reply:
Ohh yes, you were absolutely correct. I missed that case when I scanned through, my apologies.